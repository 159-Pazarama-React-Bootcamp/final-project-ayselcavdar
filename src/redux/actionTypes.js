const LOGIN_START = 'LOGIN_START';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_FAIL = 'LOGIN_FAIL';

const LOGOUT_START = 'LOGOUT_START';
const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
const LOGOUT_FAIL = 'LOGOUT_FAIL';

const SET_USER = 'SET_USER';

const GET_APPLICATIONS_START = 'GET_APPLICATIONS_START';
const GET_APPLICATIONS_SUCCESS = 'GET_APPLICATIONS_SUCCESS';
const GET_APPLICATIONS_FAIL = 'GET_APPLICATIONS_FAIL';

const DELETE_APPLICATION_START = 'DELETE_APPLICATION_START';
const DELETE_APPLICATION_FAIL = 'DELETE_APPLICATION_FAIL';
const DELETE_APPLICATION_SUCCESS = 'DELETE_APPLICATION_SUCCESS';

const CREATE_APPLICATION_START = 'CREATE_APPLICATION_START';
const CREATE_APPLICATION_SUCCESS = 'CREATE_APPLICATION_SUCCESS';
const CREATE_APPLICATION_FAIL = 'CREATE_APPLICATION_FAIL';

const EDIT_APPLICATION_START = 'EDIT_APPLICATION_START';
const EDIT_APPLICATION_FAIL = 'EDIT_APPLICATION_FAIL';
const EDIT_APPLICATION_SUCCESS = 'EDIT_APPLICATION_SUCCESS';


export {
  LOGIN_START,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_START,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL,
  SET_USER,
  CREATE_APPLICATION_FAIL,
  CREATE_APPLICATION_START,
  CREATE_APPLICATION_SUCCESS,
  GET_APPLICATIONS_FAIL,
  GET_APPLICATIONS_START,
  GET_APPLICATIONS_SUCCESS,
  DELETE_APPLICATION_START,
  DELETE_APPLICATION_SUCCESS,
  DELETE_APPLICATION_FAIL,
  EDIT_APPLICATION_FAIL,
  EDIT_APPLICATION_START,
  EDIT_APPLICATION_SUCCESS,
};
