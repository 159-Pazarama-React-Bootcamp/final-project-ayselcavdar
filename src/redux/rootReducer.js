import {combineReducers} from 'redux';
import { authReducer } from './reducers/authReducer';
import applicationReducer from './reducers/applicationReducer';

const reducers = {
  user: authReducer,
  data: applicationReducer,
};

export const rootReducer = combineReducers(reducers);