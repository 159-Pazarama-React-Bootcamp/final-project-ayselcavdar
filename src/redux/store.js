import {createStore, applyMiddleware} from 'redux';
import {rootReducer} from './rootReducer';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middleware = [thunk];
// state değişikliklerini takibe yarayan anlık loglar sunan logger modulü,
// yalnızca development ortamı için uygulanmıştır
if(process.env.NODE_ENV === 'development'){
    middleware.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middleware));