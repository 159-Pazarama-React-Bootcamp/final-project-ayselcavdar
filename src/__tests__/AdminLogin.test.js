/**
 * @jest-environment jsdom
 */
import render from 'react-test-renderer';
import AdminLogin from '../components/admin/login/AdminLoginComp';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

describe('<AdminLogin /> rendering', () => {
  it('AdminLogin snapshots check', () => {
    const comp = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <AdminLogin />
        </BrowserRouter>
      </Provider>
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
