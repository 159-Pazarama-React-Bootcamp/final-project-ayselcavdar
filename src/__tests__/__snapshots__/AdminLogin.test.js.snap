// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLogin /> rendering AdminLogin snapshots check 1`] = `
<div
  className="admin-login-container"
>
  <div>
    <h1>
      Adminastration
    </h1>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="email"
        >
          E-posta
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="password"
        >
          Şifre
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="confirmPassword"
        >
          Şifrenizi Onaylayın
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn"
        disabled={false}
        style={Object {}}
        type="submit"
      >
        Giriş
      </button>
    </form>
  </div>
</div>
`;

exports[`AdminLogin snapshots test 1`] = `
<div
  className="admin-login-container"
>
  <div>
    <h1>
      Adminastration
    </h1>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="email"
        >
          E-posta
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="password"
        >
          Şifre
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="confirmPassword"
        >
          Şifrenizi Onaylayın
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn"
        disabled={false}
        style={Object {}}
        type="submit"
      >
        Giriş
      </button>
    </form>
  </div>
</div>
`;
